rust/tw_cosmos_sdk/src/modules/transaction_util.rs

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright © 2017 Trust Wallet.
+
+use tw_coin_entry::coin_context::CoinContext;
+use tw_coin_entry::error::prelude::*;
+use tw_coin_entry::modules::transaction_util::TransactionUtil;
+use tw_encoding::base64;
+use tw_encoding::base64::STANDARD;
+use tw_encoding::hex::encode;
+use tw_hash::sha2::sha256;
+
+pub struct CosmosTransactionUtil;
+
+impl TransactionUtil for CosmosTransactionUtil {
+    fn calc_tx_hash(&self, coin: &dyn CoinContext, encoded_tx: &str) -> SigningResult<String> {
+        Self::calc_tx_hash_impl(coin, encoded_tx)
+    }
+}
+
+impl CosmosTransactionUtil {
+    fn calc_tx_hash_impl(_coin: &dyn CoinContext, encoded_tx: &str) -> SigningResult<String> {
+        let tx = base64::decode(encoded_tx, STANDARD)?;
+
+        let tx_hash = sha256(&tx);
```

Comment:
`tw_cosmos_sdk` is used by many Cosmos-based chains. There is `Greenfield` that uses `keccak256` hash instead of `sha256`. The type of hash is provided in `Cosmos::Proto::SigningInput::tx_hasher`.
Most likely we should allow an external user to specify some parameters for the transaction to be hashed correctly.
I'll think about how to better implement it. Brb in one/two days once I have something in mind

Reply:
We can choose not to regard Greenfield as a Cosmos chain.
https://github.com/trustwallet/wallet-core/blob/af67cae30f132ec139b72b4b93d2b336b84a8c49/registry.json#L4184

And implement a GreenfieldTransactionUtil for Greenfield

Reply:
I think it's fine to use `CosmosContext::default_hasher()`, so please refactor `CosmosTransactionUtil` to have a generic type parameter like:
```
struct CosmosTransactionUtil<Context: CosmosContext> { todo };
```

Reply:
Thank you for your advice. I didn't realize that we already have the StandardCosmosContext. I've adjusted the code according to your suggestion.

I was planning to make calcTxHash support Greenfield as well, but I discovered that there might be some changes in Greenfield since the wallet core started supporting it. I noticed this because the links in the test cases are now returning 404 errors.

https://github.com/trustwallet/wallet-core/blob/f703596e702cea94f23017099088a856c36b52dd/tests/chains/Greenfield/TransactionCompilerTests.cpp#L23
https://github.com/trustwallet/wallet-core/blob/f703596e702cea94f23017099088a856c36b52dd/tests/chains/Greenfield/TWAnySignerTests.cpp#L15

Therefore, I won't be supporting it in this pull request.

Could you please review the code for me again?